src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -444,6 +444,99 @@ def create_loops(body):
     omplooptrans.validate(loops[0])
 
 
+def test_omplooptrans_apply_firstprivate(fortran_reader, fortran_writer):
+    ''' Test applying the OMPLoopTrans in cases where a firstprivate
+    clause it is needed to generate functionally equivalent code than
+    the starting serial version. '''
+
+    # Example with a conditional write and a OMPParallelDoDirective
+    psyir = fortran_reader.psyir_from_source(f'''
+        subroutine my_subroutine()
+            integer :: ji, jj, jk, jpkm1, jpjm1, jpim1, scalar1, scalar2
+            real, dimension(10, 10, 10) :: zwt, zwd, zwi, zws
+            do jk = 2, jpkm1, 1
```

Comment:
Initial value for `scalar1` please (in case we ever get as far as doing analysis that tells us it is never set).

Reply:
Done